modules/raft/src/main/java/org/apache/ignite/internal/raft/RaftGroupServiceImpl.java

```diff
@@ -608,16 +607,20 @@ private <R extends NetworkMessage> void sendWithRetry(
                                     err == null ? null : err.getMessage());
                         }
 
-                        if (err != null) {
-                            handleThrowable(err, peer, request, requestFactory, stopTime, fut, retryCount);
-                        } else if (resp instanceof ErrorResponse) {
-                            handleErrorResponse((ErrorResponse) resp, peer, request, requestFactory, stopTime, fut, retryCount);
-                        } else if (resp instanceof SMErrorResponse) {
-                            handleSmErrorResponse((SMErrorResponse) resp, fut);
-                        } else {
-                            leader = peer; // The OK response was received from a leader.
-
-                            fut.complete((R) resp);
+                        try {
+                            if (err != null) {
+                                handleThrowable(err, peer, request, requestFactory, stopTime, fut, retryCount);
+                            } else if (resp instanceof ErrorResponse) {
+                                handleErrorResponse((ErrorResponse) resp, peer, request, requestFactory, stopTime, fut, retryCount);
+                            } else if (resp instanceof SMErrorResponse) {
+                                handleSmErrorResponse((SMErrorResponse) resp, fut);
+                            } else {
+                                leader = peer; // The OK response was received from a leader.
+
+                                fut.complete((R) resp);
+                            }
+                        } catch (Throwable e) {
```

Comment:
It's a remnant of one of the previous approaches. Maybe it's not needed now, but I would prefer to leave it, because currently any of the exceptions that happen *inside* the error handling code will be ignored